File: tile.cpp
#include "tile_state.cpp"

// Make a class that represents a tile for the tic tac toe board
class Tile
{
    private:
        tile_state state; 
        int x;
        int y;

    public:
        // Constructor
        Tile()
        {
            state = N;
        }

        Tile(int x, int y, tile_state state)
        {
            this->x = x;
            this->y = y;
            this->state = state;
        }

        tile_state get_state()
        {
            return state;
        }

        int get_x()
        {
            return x;
        }

        int get_y()
        {
            return y;
        }

        void set_state(tile_state state)
        {
            this->state = state;
        }

        void set_location(int x, int y)
        {
            this->x = x;
            this->y = y;
        }
};

File: Tile.java
import javax.swing.*;
import java.awt.*;

/*
 * Author: Riley Radle
 * Description: 
 *      This class represents each tile 
 *      on the tic tac toe board.  Each tile has three
 *      attributes.  Each tile is a JButton.  Their event
 *      is handled in Driver.java.
 */
public class Tile extends JButton
{
    /** The state of the Tile (X, O, or N) */
    private int state;

    /**
     * Constructor sets up the state of this Tile
     *
     * @param state : X, O, or N
     */
    public Tile(int state) 
    {
        this.state = state;
    }

    /**
     * @return : The state of this Tile
     */
    public int getState() 
    {
        return state;
    }

    /**
     * Set the state of this Tile
     * 
     * @param state : X, O, or N
     */
    public void setState(int state) 
    {
        this.state = state;
        setGraphics(state);
    }

    /**
     * Set the graphics of this Tile based on its state
     * 
     * @param state : X, O, or N
     */
    private void setGraphics(int state)
    {
        if (state != Data.PLAYER_N)
        {
            String text = state == Data.PLAYER_X ? "X" : "O";
            this.setText(text);
            this.setFont(new Font("Calibri", Font.PLAIN, 120));
        }
        else
        {
            this.setText("");
        }
    }
}
